pkg/tagger/tagstore.go

```diff
@@ -55,31 +59,45 @@ func (s *tagStore) processTagInfo(info *collectors.TagInfo) error {
 	}
 
 	// TODO: check if real change
-	s.storeMutex.RLock()
+	s.storeMutex.Lock()
+	defer s.storeMutex.Unlock()
 	storedTags, exist := s.store[info.Entity]
-	s.storeMutex.RUnlock()
-	if exist == false {
+	if !exist {
 		storedTags = &entityTags{
 			lowCardTags:  make(map[string][]string),
 			highCardTags: make(map[string][]string),
 		}
+		s.store[info.Entity] = storedTags
 	}
 
 	storedTags.Lock()
+	defer storedTags.Unlock()
 	storedTags.lowCardTags[info.Source] = info.LowCardTags
 	storedTags.highCardTags[info.Source] = info.HighCardTags
 	storedTags.cacheValid = false
-	storedTags.Unlock()
 
-	if exist == false {
-		s.storeMutex.Lock()
-		s.store[info.Entity] = storedTags
-		s.storeMutex.Unlock()
+	tagsHash := computeTagsHash(info)
```

Comment:
From what I understand, we compute a hash at the tagInfo level (only one source), while the tagger can aggregate several sources (docker + ecs or docker + kubelet for example)